restli-int-test/src/test/java/com/linkedin/restli/examples/TestParseqBasedFluentClientApi.java

```diff
@@ -644,6 +659,200 @@ public void testAssociateResourceBatchPartialUpdate() throws Exception
+    CompletableFuture<Long> createFuture = result.toCompletableFuture();
+    Long createdId = createFuture.get(5000, TimeUnit.MILLISECONDS);
+    Assert.assertTrue(subgreetingsFluentClient.get(createdId).toCompletableFuture().get(5000, TimeUnit.MILLISECONDS).getMessage().equals(msg));
+
+    // batch create
+    String msg1 = Double.toString(Math.random());
+    String msg2 = Double.toString(Math.random());
+    CompletionStage<List<CreateIdStatus<Long>>>
+        batchCreateStage = subgreetingsFluentClient.batchCreate(Arrays.asList(getGreeting(msg1), getGreeting(msg2)));
+    CompletableFuture<List<CreateIdStatus<Long>>> batchCreateFuture = batchCreateStage.toCompletableFuture();
+    List<CreateIdStatus<Long>> createdList = batchCreateFuture.get(5000, TimeUnit.MILLISECONDS);
+    CompletionStage<Map<Long, EntityResponse<Greeting>>> batchGetStage =
+        subgreetingsFluentClient.batchGet(createdList.stream().map(CreateIdStatus::getKey).collect(Collectors.toSet()));
+
+    Map<Long, EntityResponse<Greeting>> entities = batchGetStage.toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
+    Assert.assertEquals(entities.size(), 2);
+  }
+
+  /**
+   * Test {@link com.linkedin.restli.examples.greetings.server.SimpleResourceUnderCollectionResource}
+   * A complete set of request tests were tested in {@link TestSimpleResourceHierarchy}
+   */
+  @Test public void testSubResource_oneLayerPathKey() throws Exception
+  {
+    // Get
+    SubsubgreetingFluentClient subsubClient = new SubsubgreetingFluentClient(_parSeqRestliClient, _parSeqUnitTestHelper.getEngine());
+    Greeting greeting = subsubClient.withSubgreetingsId(1L).get().toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
+    Assert.assertEquals(greeting.getId().longValue(), 10L);
+
+    // Update
+    Greeting updateGreeting = new Greeting();
+    updateGreeting.setMessage("Message1");
+    updateGreeting.setTone(Tone.INSULTING);
+    updateGreeting.setId(1L);
+    subsubClient.update(updateGreeting).toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
+    Assert.assertEquals(subsubClient.get().toCompletableFuture().get(5000, TimeUnit.MILLISECONDS).getTone(), Tone.INSULTING);
+
+    // Partial Update
+    Greeting partialUpdateGreeting = new Greeting();
+    partialUpdateGreeting.setMessage("Message1");
+    partialUpdateGreeting.setTone(Tone.SINCERE);
+    partialUpdateGreeting.setId(1L);
+    PatchRequest<Greeting> patch = PatchGenerator.diffEmpty(partialUpdateGreeting);
+    subsubClient.partialUpdate(patch).toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
+    Assert.assertEquals(subsubClient.get().toCompletableFuture().get(5000, TimeUnit.MILLISECONDS).getTone(), Tone.SINCERE);
+
+    // Delete
+    subsubClient.delete().toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
+    try
+    {
+      subsubClient.get().toCompletableFuture().get(5000, TimeUnit.MILLISECONDS);
+      Assert.fail("Should fail");
```

Comment:
"Sub resource call without path key should fail."

Reply:
resolved